### PR TITLE
Apply theme colours to branding text by default. #4501

### PR DESCRIPTION
**Pull request by Shavindra**
## Summary

Issue #4501 - Branding Config: Text color shown as white when not configured

## Technical notes
- Updated BannerSettings component to retrieve the default values using `getComputedStyle`.
- `ColorInput` component would use the `:default-value` if `:value === null`
- Same default `--banner-text-color` should apply to Header, Footer and Banner consent.

## Steps to test

1. Default settings
    1. Navigate to Configuration > Global Settings > Branding
    2. Tick ALL or each of the `Show banner in header`, `Show banner in footer` and `Show Consent Banner on Login Screen`
    3. Light theme - `Text Color` option and should highlight `#141419`
    4. Dark theme - `Text Color` option and should highlight `#B6B6C2`
    5. Applying the setting should ensure selected colour is set
    6. Colour should remain in default of the theme (when switching themes), if user didn't change the colour and only edit the text.
    7. 3 Banners `Header`,`Footer` and `Banner Consent` text colours are set independently.

2. Legacy to Default settings
    1. If have legacy UI navigate to Settings > ui-banner.
    2. Update the banner settings json to be `{}`
    3. Test again with the 1. Default settings steps. UI Should work the same way

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
This does what's asked and fixes the problem, however it's highlighted another bug. 

After creating a banner without touching either text or background colours, switching between light and dark modes shows the default colour text that was shown at the time of creation, rather than how the background reacts to the current mode. It looks like a specific colour is set for text whilst the background colour is either set as null or not set at all. I guess when the banner's are shown if the value is null is picks whatever's best given the current theme.

Bottom line - I think we should show the theme colours... but if the user hasn't change it to avoid setting the value altogether or specifically set as null. Hopefully that will fix the issue

**Comment by vincent99** [comment]
I mentioned not using SCSS at runtime and the burning the current color into the settings in slack last week:

> Vincent Fiduccia  11 days ago
> Runtime stuff normally operates only on CSS variables (--something: #123456;), not the SCSS (but those css variables are often populated by a sass value in the theme file) or dynamic lookups in JS.
> 
> That way the applied theme is only determined by the variables that are set, and everything updates automatically if the selected theme (light, dark, auto) changes by just updating the classes on the body tag.  Light is the base/default so it's always applied, dark overrides things as appropriate, and I'd expect the custom user theme stuff to add additional overrides on top for what they configured.
> 
> Shavin Fonseka  8 days ago
> with scss :export you can import SCSS variables directly in to Javascript. I've tested it switching themes and it works.
> 
> Vincent Fiduccia  8 days ago
> I know you _can_, but am saying you _shouldn't_.  You don't need to lookup what the SCSS says, you can just set it to var(--primary-text) or whatever it is.
> The runtime should only look at CSS variables, because we're eventually going to do 1st/3rd party addons.  They might want to change the effective CSS (e.g. provide their own neon green & orange theme), but won't have access to the build-time SCSS.
> 
> Vincent Fiduccia  8 days ago
> Also the editor defaults to 'burning in' the light theme text color (#141419) into the setting value instead of saving nothing and then using it as a default, and defaults to nothing for the background.. so the combination of that is black(ish) on black if you later switch to dark theme.  Both color pickers should default to saving nothing and applying the default on display, so that if we change the theme colors it updates you automatically unless you've customized it.

File: mixins/brand.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ export default {
       return setting?.value;
     },
 
+    themeVars() {
+      return getCssThemeVarsFromDocument(`theme-${ this.$store.getters['prefs/theme'] }`);
```
Comment (nwmac) [inline mixins/brand.js:24]: This seems like a lot of effort to get the css vars - we already do this in `brand.vue` to get a single value, e.g.

`let uiColor = getComputedStyle(document.body).getPropertyValue('--primary');`

Could we not do the same here? You could then remove the code in css-rule-parser.

We probably only need a single color var for the banner text color

**Review by nwmac — CHANGES_REQUESTED** [review]

**Comment by nwmac** [comment]
@Shavindra Can you rebase this with master - @jordojordo has added another banner on the login page = I'd suggest we have just one default color rather than `--header-text-color` and `'--header-text-color` = maybe `--banner-text-color` and use that for all banners

**Review by nwmac — CHANGES_REQUESTED** [review]
See comment

**Review by nwmac — APPROVED** [review]
LGTM
